src/core/stream/representation/representation_stream.ts

```diff
@@ -365,11 +379,20 @@ export default function RepresentationStream<TSegmentDataType>({
                                 imminentDiscontinuity: status.imminentDiscontinuity,
                                 hasFinishedLoading: status.hasFinishedLoading,
                                 neededSegments: status.neededSegments } });
-
+      const gcedPosition = Math.max(
+        0,
+        wantedStartPosition - UPTO_CURRENT_POSITION_CLEANUP);
```

Comment:
I fear that the reason behind the why of this operation could be hard to follow in the feature.

Maybe first explicitely checking the `isBufferFull` boolean, like:
```ts
let bufferRemoval = EMPTY;
if (status.isBufferFull) {
  const gcedPosition = Math.max(
        0,
        wantedStartPosition - UPTO_CURRENT_POSITION_CLEANUP);
  bufferRemoval = segmentBuffer
              .removeBuffer(0, gcedPosition)
              // eslint-disable-next-line @typescript-eslint/no-unsafe-argument
              .pipe(ignoreElements()) :
}
```
Would be easier to read, no?

Reply:
Done